codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/endpoints/EndpointResolverAdapterGenerator.kt

```diff
@@ -198,8 +213,38 @@ class EndpointResolverAdapterGenerator(
                 continue
             }
 
-            inputContextParams[paramName]?.let {
-                writer.write("builder.#L = input.#L", paramDefaultName, it.defaultName())
+            // Look at input params
+            val inputParam = inputContextParams[paramName]
+            if (inputParam != null) {
+                writer.write("builder.#L = input.#L", paramDefaultName, inputParam.defaultName())
+                continue
+            }
+
+            // Look at operation params
+            val operationParam = operationContextParams?.get(paramName)
+            if (operationParam != null) {
+                val opInputShape = ctx.model.expectShape(op.inputShape)
+                val inputSymbol = ctx.symbolProvider.toSymbol(opInputShape)
+
+                if (inputContextParams.isEmpty()) {
+                    // This will already be rendered in the block if inputContextParams is not empty
+                    writer.write("val input = request.context[#T.OperationInput] as #T", RuntimeTypes.HttpClient.Operation.HttpOperationContext, inputSymbol)
+                }
```

Comment:
**Question**: The code to render this version of fetching the input is different from the code above to render it:

```kotlin
if (inputContextParams.isNotEmpty()) {
    writer.addImport(RuntimeTypes.Core.Collections.get)
    writer.write("@Suppress(#S)", "UNCHECKED_CAST")
    val opInputShape = ctx.model.expectShape(op.inputShape)
    val inputSymbol = ctx.symbolProvider.toSymbol(opInputShape)
    writer.write("val input = request.context[#T.OperationInput] as #T", RuntimeTypes.HttpClient.Operation.HttpOperationContext, inputSymbol)
}
```

Why are the explicit import and the cast suppression necessary there but not here?

Reply:
I think they are necessary, let me add them